src/taipy/core/_repository/_repository_adapter.py

```diff
@@ -0,0 +1,29 @@
+# Copyright 2022 Avaiga Private Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+# the License. You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+from importlib import util
+from typing import Any, Optional, Union
+
+from src.taipy.core._repository import _FileSystemRepository
+
+
+class _RepositoryAdapter:
+    _TAIPY_ENTERPRISE_MODULE = "taipy.enterprise"
+    _TAIPY_ENTERPRISE_CORE_MODULE = _TAIPY_ENTERPRISE_MODULE + ".core"
+
+    _REPOSITORY_MAP = {"default": _FileSystemRepository}
+
+    @classmethod
+    def select_base_repository(cls) -> Optional[Union[_FileSystemRepository, Any]]:
+        return cls._REPOSITORY_MAP.get("default")
```

Comment:
Do we need to add something like:
```suggestion
        if cls._using_enterprise():
                return _load_fct(
                    cls._TAIPY_ENTERPRISE_CORE_MODULE + "repository.adapter", 
                    "RepositoryAdapter",
                )()
        return cls._REPOSITORY_MAP.get("default")
```

Depending on the config, if we are using SQL for instance, we want the adapter.select_base_repository() to return _SQLRepository, no?

Reply:
That's true. Something I missed while refactoring the code from the poc.